designer/client/src/components/modals/ActionParameter.tsx

```diff
@@ -0,0 +1,49 @@
+import { ExpressionLang } from "../graph/node-modal/editors/expression/types";
+import React, { useCallback } from "react";
+import { FieldLabel } from "../graph/node-modal/FieldLabel";
+import { getValidationErrorsForField } from "../graph/node-modal/editors/Validators";
+import { ActionNodeParameters, ActionParameterConfig } from "../../types/action";
+import { NodesDeploymentData } from "../../http/HttpService";
+import { NodeValidationError } from "../../types";
+import { default as EditableEditor } from "../graph/node-modal/editors/EditableEditor";
+
+interface Props {
+    nodeName: string;
+    propertyName: string;
+    propertyConfig: ActionParameterConfig;
+    nodesData: NodesDeploymentData;
+    onChange: <K extends keyof ActionNodeParameters["parameters"]>(
+        nodeId: string,
+        property: K,
+        newValue: ActionNodeParameters["parameters"][K],
+        defaultValue?: ActionNodeParameters["parameters"][K],
+    ) => void;
+    errors: NodeValidationError[];
+}
+
+export function ActionParameter(props: Props): JSX.Element {
+    const { nodeName, propertyName, propertyConfig, errors, nodesData, onChange } = props;
+
+    const current = nodesData[nodeName][propertyName] || "";
```

Comment:
Maybe it's worth renaming it to currentNodePropertyName for readability sake?

Reply:
Fixed (removed whole nodesData, it is not necessary here).